rs/boundary_node/rate_limits/canister/interface.did

```diff
@@ -0,0 +1,111 @@
+
+
+// Input structure for defining a rule with mandatory fields within a config
+type InputRule = record {
+  incident_id: IncidentId; // Unique identifier for the incident
+  rule_raw: blob;          // Raw rule data (in binary format), expected to be a valid json object
+  description: text;       // Textual description of the rule
+};
+
+// Output structure for rules
+// Optional fields rule_raw and description may remain hidden while the rule is under confidentiality restrictions
+type OutputRule = record {
+  id: RuleId;                        // Unique identifier for the rule
+  rule_raw: opt blob;                // Raw rule data (in binary format), expected to be a valid json object, none if the rule is currently confidential
+  description: opt text;             // Textual description of the rule, none if the rule is currently confidential
+  disclosed_at: opt Timestamp;       // Timestamp when the rule was disclosed, none if the rule is still confidential
+};
+
+type OutputConfig = record {
+  rules: vec OutputRule;
+};
+
+// Response structure for returning the requested configuration and associated metadata
+type OutputConfigResponse = record {
+    version: Version;         // Version of the configuration
+    active_since: Timestamp;  // Time when this configuration was added (became active)
+    config: OutputConfig;     // Contains the list of rules
+};
+
+// Verbose details of an individual rule
+// Optional rule_raw and description fields are for restricted publicly viewing access
+type OutputRuleMetadata = record {
+  id: RuleId;                      // Unique identifier for the rule
+  rule_raw: opt blob;              // Raw rule data (binary format), expected to be a valid json object, none if the rule is currently confidential
+  description: opt text;           // Textual description of the rule, none if the rule is currently confidential
+  disclosed_at: opt Timestamp;     // Timestamp when the rule was disclosed, none if the rule is still confidential
+  added_in_version: Version;       // Version when the rule was added (became active)
+  removed_in_version: opt Version; // Version when the rule was deactivated (removed), none if the rule is still active
+};
+
+type GetRuleByIdResponse = variant {
+  Ok: OutputRuleMetadata;
+  Err: text;
+};
+
+type GetConfigResponse = variant {
+  Ok: OutputConfigResponse;
+  Err: text;
+};
+
+type DiscloseRulesResponse = variant {
+  Ok;
+  Err: text;
+};
+
+type OverwriteConfigResponse = variant {
+  Ok;
+  Err: text;
+};
+
+type DiscloseRulesByRuleIdsResponse = variant {
+  Ok;
+  Err: text;
+};
+
+type DiscloseRulesByIncidentIdResponse = variant {
+  Ok;
+  Err: text;
+};
+
+type GetRulesByIncidentIdResponse = variant {
+  Ok: vec RuleId;
+  Err: text;
+};
+
+// Configuration containing a list of rules that replaces the current configuration
+type InputConfig = record {
+  schema_version: SchemaVersion; // schema version used to serialized the rules
+  rules: vec InputRule;
+};
```

Comment:
Should it be possible when specifying rules to also specify whether they are public or not? Or do you have to basically set the config, and then in another call disclose rules? I guess what I'm asking is "what does a normal user flow look like?"

Reply:
The expected flow is:
1. Each newly added it is confidential by default
2. To open the rule for public viewing a separate call to `disclose_()` function is needed
We could indeed add an additional field `policy` with `private/public`, but it seems to be unnecessary. If needed it can be added.